### PR TITLE
Remove geocoq from CI

### PR DESCRIPTION
**Pull request by SkySkimmer**
It has been allow_failure for over a month.

I just removed the yml job, not sure if I should also remove the makefile target, script and basic-overlay entry.

### CODE REVIEW

**Comment by Zimmi48** [comment]
Removing is a good thing to save computational resources over a (currently) useless job. However, no need to clean it up from our infrastructure just yet as we might add it back if fixed. What about commenting out the lines in `gitlab-ci.yml` instead of removing them to highlight this?

BTW, are GeoCoq maintainers aware that we have this issue with their project currently not building? We should probably have opened an issue, or at least @-mention their maintainers here (@jnarboux).

**Comment by SkySkimmer** [comment]
>What about commenting out the lines in gitlab-ci.yml instead of removing them to highlight this?

I don't like dead code comments.

**Comment by maximedenes** [comment]
> What about commenting out the lines in `gitlab-ci.yml` instead of removing them to highlight this?

It doesn't seem harder to revert rather than uncomment. And it looks less messy than leaving commented code, but maybe I'm missing the point.

**Comment by ejgallego** [comment]
I agree with @Zimmi48 , the job should be fully cleaned up or just commented out with a comment.

**Comment by SkySkimmer** [comment]
fully removed

File: dev/ci/nix/default.nix [hunk 1]
```diff
@@ -104,7 +104,6 @@ let projects = {
   fiat_crypto = callPackage ./fiat_crypto.nix {};
   flocq = callPackage ./flocq.nix {};
   formal-topology = callPackage ./formal-topology.nix {};
-  GeoCoq = callPackage ./GeoCoq.nix {};
```
Comment (Zimmi48) [inline dev/ci/nix/default.nix:107]: There is still a `GeoCoq.nix` file in the folder (although I honestly do not care if the cleanup goes as far as the `dev/ci/nix` folder, which is entirely untested anyway).

**Review by Zimmi48 — APPROVED** [review]
@coqbot merge now

**Comment by ejgallego** [comment]
There is still the question about how to best easily test stuff that is not in CI but we may need to check, for example see #12674 .

I have some ideas as (some of us) discussed in the CUDW, so get in touch if you are interested on it.

**Pull request merged**